app/controllers/username_controller.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class UsernameController < ApplicationController
```

Comment:
I don't have the full context but wouldn't it be better to have a user end point rather than a username endpoint. We can restrict the attributes that can be updated to just username if that's what is desired.

Reply:
We don't really have the full context yet either - right now the username is the only thing that can be updated, but in the future if other stuff can be updated we don't know if it would happen on the same page or other pages.

I decided to go with a username controller instead of a more generic user controller as it's only one field that can be updated on this page, and it redirects to the dashboard instead of rendering the edit page again - I feel like with changing email or something like that you'd almost certainly want a different behaviour, and thus a different update method.

I don't think it'd be too tricky to change in the future, though - this isn't exactly a page that people will be bookmarking!